src/components/pages/myLoan/component.tsx

```diff
@@ -7,103 +7,108 @@ import {Row, Col} from "../../lib";
 import Withdrawal from "./withdrawals";
 import RepaymentStatus from "./repaymentStatus";
-import {RepaymentManager, TermsContract} from "../../../utils/contractData";
 import contractAddresses from "../../../config/ines.fund.js";
-import {getContractInstance} from "../../../utils/getDeployed";
 import {
-    contractGetPastEvents,
-    contractMethodCall,
     getInjectedAccountAddress,
-    getNetworkId
+    prepBigNumber
 } from "../../../utils/web3Utils";
-import getWeb3 from "../../../utils/getWeb3";
+import { getDeployedFromConfig } from "../../../utils/getDeployed";
+import { getTokenDetailsFromAddress } from '../../../utils/paymentToken';
+import { shares, released, releaseAllowance, totalPaid, PaymentReceivedEvent, PaymentReleasedEvent } from '../../../utils/repaymentManager';
+import { getPrincipalDisbursed, getPrincipalToken } from '../../../utils/termsContract';
 
 interface MyLoanState {
+    paymentToken: object,
     principalDisbursed: string;
+    shares: string;
     totalPaid: string;
     releaseAllowance: string;
     withdrawals: object;
+    repayments: object;
 }
 interface MyLoanProps extends RouteComponentProps<any> {}
 class MyLoan extends React.Component<MyLoanProps, MyLoanState> {
     state = {
+        paymentToken: null,
         principalDisbursed: "",
+        shares: "",
         totalPaid: "",
         releaseAllowance: "",
-        withdrawals: null
+        withdrawals: null,
+        repayments: null
     };
 
     componentDidMount = async () => {
         try {
-            const networkId = await getNetworkId();
-            const repaymentManagerAddress = contractAddresses[networkId]["RepaymentManager"];
-            const termsContractAddress = contractAddresses[networkId]["TermsContract"];
-
-            const repaymentManagerInstance = await getContractInstance(
-                RepaymentManager.abi,
-                repaymentManagerAddress
-            );
-
-            const termsContractInstance = await getContractInstance(
-                TermsContract.abi,
-                termsContractAddress
-            );
+            
+            const termsContractInstance = await getDeployedFromConfig('TermsContract', contractAddresses);
+            const repaymentManagerInstance = await getDeployedFromConfig('RepaymentManager', contractAddresses);
+            const paymentToken = await getTokenDetailsFromAddress(await getPrincipalToken(termsContractInstance));
 
             // Note: principal disbursed and total paid will return zero when the loan is not started
-            const principalDisbursed = await contractMethodCall(
-                termsContractInstance,
-                "getPrincipalDisbursed"
-            );
-            const totalPaid = await contractMethodCall(repaymentManagerInstance, "totalPaid");
+            const principalDisbursed = await getPrincipalDisbursed(termsContractInstance);
+            const _totalPaid = await totalPaid(repaymentManagerInstance);
```

Comment:
"_" indicates a local variable?

Reply:
there is a conflict with an imported function from the `repaymentManger.js`